src/sql/base/parts/editableDropdown/browser/dropdownTree.ts

```diff
@@ -124,10 +125,24 @@ export class DropdownController extends TreeDefaults.DefaultController {
 		return response;
 	}
 
+	public onKeyDown(tree: tree.ITree, event: IKeyboardEvent): boolean {
+		// The enter key press is handled on key up by our base class (DefaultController) but
+		// we want to stop it here because we know we're going to handle it (by selecting the item)
+		// and letting it propagate up means that other controls may incorrectly handle it first
+		// if they're listening to onKeyDown
+		const response = super.onKeyDown(tree, event);
+		if (event.keyCode === KeyCode.Enter) {
```

Comment:
Seems like you are stopping twice? do we need this  one if you stop it later.

Reply:
What do you mean we stop it later? Are you talking about in the onEnter? That one is triggered on key up so is a separate event. The comment explains that I do this here because everywhere else we've been using key down and so if I just do it there then the key down is still handled in those other places (like the dialog) and we want to specifically stop that.

---
In reply to: [339764920](https://github.com/microsoft/azuredatastudio/pull/8050#discussion_r339764920) [](ancestors = 339764920)